x-pack/plugins/observability_solution/infra/public/components/asset_details/template/flyout.tsx

```diff
@@ -53,7 +55,7 @@ export const Flyout = ({
       data-component-name={ASSET_DETAILS_FLYOUT_COMPONENT_NAME}
       data-asset-type={asset.type}
     >
-      {loading ? (
+      {loading || metadataLoading ? (
```

Comment:
This will block the loading of components in the flyout that doesn't depend on metadata.

https://github.com/user-attachments/assets/ff890161-2c2f-47ff-849c-f1c67e0d65ac

This is not necessarily part of this PR, but how about removing this `InfraLoadingPanel` and passing `loading` and `metadataLoading` via props to `FlyoutHeader`? This way we can have a loading spinner that doesn't block the loading rest of the flyout.

For the full page view, we do [this](https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability_solution/infra/public/components/asset_details/template/page.tsx#L86)

Reply:
Thanks for spotting that! I changed it to use the loading spinner inside `FlyoutHeader ` like on the page view: 


https://github.com/user-attachments/assets/1c17c775-0657-4db8-9e91-9faa8ddbc9a0

Reply:
It looks great now! thanks.